### ISSUE TITLE
Avoid propagating passed in text query directly to worker nodes

### ISSUE THREAD

**Issue by samay-sharma** [citusdata/citus#356]
@anarazel brought up the following in context of DDL command propagation:

1) The query string can potentially contain multiple SQL statements and such. In that case just sending it to the workers as-is is obviously wrong.

2) The original query depends on the search path and such on the primary, something that's potentially going to be different on workers. We should never just send query text ahead as-is.

**Comment by ozgune** [comment]
@samay-sharma: Is this the issue we opened for deparsing these commands on the master node?

If so, we made a few changes in #653 and #678 that may relate to this issue. If we deparsed these commands on the master, we could then avoid passing down schema name as an extra parameter.

**Comment by samay-sharma** [comment]
Yes, what I had in mind was something similar to ReplicateGrantStmt where we construct the GRANT statement from the parsetree.

**Comment by saicitus** [comment]
+1, came up in a customer engagement. Not a blocking issue at all. But wanted to document it here.
